packages/server/models/Post.js

```diff
@@ -0,0 +1,25 @@
+const mongoose = require("mongoose");
+
+const PostSchema = new mongoose.Schema({
+    title: {
+        type: String,
+        required: true,
+    },
+    body: {
+        type: String,
+        required: true,
+    },
+    isDraft: {
+        type: Boolean,
+        required: true,
+    },
+    createdAt: {
+        type: Date,
+        default: Date.now,
+    },
+    publishedAt: {
+        type: Date
+    }
```

Comment:
These aren't setup naturally, createdAt and updatedAt could be done with timestamps.
- [mongoose docs](https://mongoosejs.com/docs/guide.html#timestamps)
- [simple link](https://masteringjs.io/tutorials/mongoose/timestamps)

Reply:
For published at we could probably do a [pre save hook](https://mongoosejs.com/docs/middleware.html#order) using [`getChanges`](https://mongoosejs.com/docs/api.html#document_Document-getChanges) to see if `isDraft` is true.